themes/default/content/blog/next-level-iac-pulumi-automation-api/index.md

```diff
@@ -0,0 +1,357 @@
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ServicePayload'
+      responses:
+        '200':
+          description: Infrastructure successfully created
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  url:
+                    type: string
+                    description: URL of the created infrastructure
+        '400':
+          description: Bad request if the input payload is incorrect
+        '500':
+          description: Internal server error
+    delete:
+      summary: Destroy Infrastructure
+      operationId: destroyInfrastructure
+      parameters:
+        - in: query
+          name: name
+          schema:
+            type: string
+          required: true
+          description: Name of the service
+        - in: query
+          name: org
+          schema:
+            type: string
+          required: true
+          description: Organization in Pulumi
+        - in: query
+          name: project
+          schema:
+            type: string
+          required: true
+          description: Project name in Pulumi
+        - in: query
+          name: stack
+          schema:
+            type: string
+          required: true
+          description: Stack name in Pulumi
+        - in: query
+          name: region
+          schema:
+            type: string
+          required: true
+          description: AWS region where the service will be deployed
+      responses:
+        '200':
+          description: Infrastructure successfully destroyed
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    description: Confirmation message of the destruction
+        '400':
+          description: Bad request if the required parameters are missing
+        '500':
+          description: Internal server error
+components:
+  schemas:
+    ServicePayload:
+      type: object
+      required:
+        - name
+        - org
+        - project
+        - stack
+        - region
+      properties:
+        name:
+          type: string
+          description: Name of the service
+        org:
+          type: string
+          description: Organization in Pulumi
+        project:
+          type: string
+          description: Project name in Pulumi
+        stack:
+          type: string
+          description: Stack name in Pulumi
+        region:
+          type: string
+          description: AWS region where the service will be deployed
+```
+
+Navigate to the [Swagger Editor](https://editor.swagger.io/) and paste the above OpenAPI specification to see how it looks like.
+
+![Swagger Editor](swagger.png)
+
+## Conclusion
+
```

Comment:
The term infrastructure as a service might be overloaded since clouds like aws are known as IaaS.